src/google_trace_exporter_client.rs

```diff
@@ -60,7 +62,18 @@ impl GcpCloudTraceExporterClient {
                     display_name: Some(Self::truncatable_string(span.name.deref(), 128)),
                     start_time: Some(prost_types::Timestamp::from(span.start_time)),
                     end_time: Some(prost_types::Timestamp::from(span.end_time)),
-                    attributes: Some(Self::convert_span_attrs(&span.attributes)),
+                    attributes: Some(Self::convert_span_attrs(
```

Comment:
It is preferable to prepare in the constructor (`new` function) converted models for resource already so we can quickly append those in `convert_span_attrs` (for that we would need to change `Self::convert_span_attrs` to `self.convert_span_attrs`, so we can have access to private structure field.

Reply:
Done. Converting `Resource` to `Vec<KeyValue>` in the constructor now and chaining it in `convert_span_attrs`.